src/orchestrator/mod.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{
+    error::Result,
+    model::{PodJob, PodResult},
+};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, future::Future, path::PathBuf};
+/// Options for sourcing compute environment images.
+pub enum ImageKind {
+    /// A published compute environment image in a container registry. Argument formatted as
+    /// `{server.com/}{name}:{tag}`. Server is optional e.g. (`alpine:latest`).
+    Published(String),
+    /// A packaged compute environment of image+tag as a tarball. Argument is the relative path of
+    /// tarball in orchestrator data directory e.g. (`path/to/image.tar.gz`).
+    Tarball(PathBuf),
+}
+/// Available states of a run.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum RunState {
+    /// Run is ongoing.
+    Running,
+    /// Run has completed successfully.
+    Completed,
+    /// Run failed with the provided error code.
+    Failed(i16),
+}
+/// Run metadata
+#[derive(Debug)]
+pub struct RunInfo {
+    /// Name given by orchestrator.
+    pub name: String,
+    /// Environment utilized.
+    pub image: String,
+    /// Time in epoch when created in seconds.
+    pub created: u64,
+    /// Time in epoch when terminated in seconds.
+    pub terminated: Option<u64>,
+    /// Environment variables set in environment.
+    pub env_vars: HashMap<String, String>,
```

Comment:
Noticed that there is no reference to pod_job. Should probably discuss the pro and cons about it.